core/shared/src/main/scala/scalaz/zio/Errors.scala

```diff
@@ -1,10 +1,18 @@
 // Copyright (C) 2017-2018 John A. De Goes. All rights reserved.
 package scalaz.zio
+import scalaz.zio.ExitResult.Cause
 
 object Errors {
 
-  final case class UnhandledError(error: Any, defects: List[Throwable] = Nil)
-      extends Exception("An error was not handled by a fiber: " + error.toString())
+  final case class FiberFailure(cause: Cause[Any]) extends Throwable {
```

Comment:
At this point, there's no need for `Errors` object to exist. I'd promote `FiberFailure` to the top level, and rename the file from `Errors.scala` to `FiberFailure.scala`.